### ISSUE TITLE
Audit `invert(powerOfTwo:within)`

### ISSUE THREAD

**Issue by jsbean** [dn-m/ArithmeticTools#65]
`invert(powerOfTwo:within)` was created to invert values within symmetrically ordered `OptionSet`:

```Swift
struct InvertibleSet: OptionSet {
    
    // inverse of d
    static let a: InvertibleSet // rawValue = 1 << 0

    // inverse of c
    static let b: InvertibleSet // rawValue = 1 << 1

    // inverse of b
    static let c: InvertibleSet // rawValue = 1 << 2

    // inverse of a
    static let d: InvertibleSet // rawValue = 1 << 3
}
```

The point of `invert` is to first decode the `rawValue` (`log2(rawValue)`), then find its inversion within the amount of options.

@brianlheim, perhaps there is a better way of implementing this (using `countTrailingZeros`).

Further, we should make a protocol with a default implementation for `OptionSet` which have this property (e.g., [`IntervalTools.Relationship`](https://github.com/dn-m/IntervalTools/blob/master/IntervalTools/Relationship.swift)).

In the [`PitchSpellingTools`](https://github.com/dn-m/PitchSpellingTools) framework, this protocol is implemented as: [`Invertible`](https://github.com/dn-m/PitchSpellingTools/blob/master/PitchSpellingTools/Invertible.swift), but there is no default implementation. This inherited by `NamedInterval`, `NamedIntervalQuality`, and `NamedIntervalOrdinal` (and therefore implemented in `AbsoluteNamedOrdinal` and `RelativeNamedOrdinal`).

Perhaps this is one for [`Collections`](https://github.com/dn-m/Collections), but we could define the `InvertibleSet` protocol as such:

```Swift
protocol InvertibleOptionSet: OptionSet {
    var inverse: Self { get }
    var optionsCount: Int { get }
}

extension InvertibleOptionSet {

    // inherit `rawValue` from `OptionSet`

    var inverse: Self {
        let originalValue = countTrailingZeros(rawValue)
        let inverseValue = (optionsCount - 1) - originalValue
        return Self(rawValue: 1 << inverseValue)!
    }
}
```

**Comment by mossheim** [comment]
Since this seems to be pretty specific to `OptionSet`, I like the idea of hiding all those implementation details behind a new protocol and using CTZ instead of defining an `invert` function.

**Comment by jsbean** [comment]
I agree completely.

**Comment by jsbean** [comment]
Now, here's a challenge: where to put this?

`ArithmeticTools` requires `Collections`.

Yet, this would require `ArithmeticTools`.

**Comment by mossheim** [comment]
Does `Collections` req `ArithmeticTools`? I wouldn't be opposed to putting this directly in `ArithmeticTools`. `OptionSet` has collection-y name, but it's really more like an enum IMO.

**Comment by jsbean** [comment]
SGTM !

**Comment by jsbean** [comment]
Does `InvertibleOptionSet` seem like a convincing name here ?

**Comment by mossheim** [comment]
Yeah, I think that's a perfect name.

**Comment by jsbean** [comment]
Closed by #66.
